src/commonControls/numberControl/NumberControl.ts

```diff
@@ -1192,6 +1297,10 @@ export class NumberControl extends Control implements InteractionModelContributo
         }
         return undefined;
     }
+
+    public getAplValidationFailedMessage() {
+        return this.props.apl.validationFailedMessage;
```

Comment:
this can also be a function. so need to evaluate it like prompt props.
Please declare the return type for all functions.  if you declare return type of string here, the compiler will tell you about the problem.